File: msg-sentiment/src/test/java/dev/number6/sentiment/ChannelMessagesSentimentComprehensionComponentIntegrationTest.java
package dev.number6.sentiment;

import dev.number6.comprehend.results.PresentableSentimentResults;
import dev.number6.message.ChannelMessages;
import dev.number6.message.ChannelMessagesHandler;
import dev.number6.sentiment.dagger.DaggerTestChannelMessageSentimentComprehensionComponent;
import dev.number6.sentiment.dagger.FakeComprehensionResultsModule;
import dev.number6.sentiment.dagger.TestChannelMessageSentimentComprehensionComponent;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;

class ChannelMessagesSentimentComprehensionComponentIntegrationTest {

    TestChannelMessageSentimentComprehensionComponent testee = DaggerTestChannelMessageSentimentComprehensionComponent.create();

    @Test
    void providesChannelHandler() {

        PresentableSentimentResults results = mock(PresentableSentimentResults.class);
        ChannelMessagesHandler handler = testee.getChannelMessagesHandler();
        FakeComprehensionResultsModule.ConfigurableResultsFunction f = (FakeComprehensionResultsModule.ConfigurableResultsFunction) testee.getResultsFunction();
        f.setPresentableSentimentResults(results);

        handler.handle(mock(ChannelMessages.class));

        FakeComprehensionResultsModule.RecordingComprehensionResultsConsumer consumer = (FakeComprehensionResultsModule.RecordingComprehensionResultsConsumer) testee.getConsumer();
        assertThat(consumer.getResultsConsumed()).hasSize(1);
        assertThat(consumer.getResultsConsumed().get(0)).isEqualTo(results);
    }
}

File: msg-sentiment/src/test/kotlin/dev/number6/sentiment/ChannelMessagesSentimentComprehensionComponentIntegrationTest.kt
package dev.number6.sentiment

import assertk.assertThat
import assertk.assertions.hasSize
import assertk.assertions.isEqualTo
import dev.number6.comprehend.results.PresentableSentimentResults
import dev.number6.message.ChannelMessages
import dev.number6.sentiment.dagger.DaggerTestChannelMessageSentimentComprehensionComponent
import dev.number6.sentiment.dagger.FakeComprehensionResultsModule
import dev.number6.sentiment.dagger.TestChannelMessageSentimentComprehensionComponent
import io.mockk.mockk
import org.junit.jupiter.api.Test

internal class ChannelMessagesSentimentComprehensionComponentIntegrationTest {
    private var testee: TestChannelMessageSentimentComprehensionComponent = DaggerTestChannelMessageSentimentComprehensionComponent.create()

    @Test
    fun providesChannelHandler() {
        val results = mockk<PresentableSentimentResults>()
        val handler = testee.getChannelMessagesHandler()
        val f = testee.getResultsFunction() as FakeComprehensionResultsModule.ConfigurableResultsFunction
        f.setPresentableSentimentResults(results)
        handler.handle(mockk())
        val consumer = testee.getConsumer() as FakeComprehensionResultsModule.RecordingComprehensionResultsConsumer
        assertThat(consumer.getResultsConsumed()).hasSize(1)
        assertThat(consumer.getResultsConsumed()[0]).isEqualTo(results)
    }
}